compiler/rustc_builtin_macros/src/asm.rs

```diff
@@ -465,6 +468,54 @@ fn expand_preparsed_asm(ecx: &mut ExtCtxt<'_>, sp: Span, args: AsmArgs) -> P<ast
         for piece in unverified_pieces {
             match piece {
                 parse::Piece::String(s) => {
+                    if let Some(idx) = s.find(".intel_syntax") {
+                        let mut end = idx + ".intel_syntax".len();
+                        if let Some(prefix_idx) = s.split_at(end).1.find("noprefix") {
```

Comment:
Directives extend to the end of the line, so you could do something along the lines of instead of manually matching the `noprefix` and possibly other modifiers:

```
let end = s[idx..].find('\n') + idx;
```